### ISSUE TITLE
Helm chart on Kubeapps is not available

### ISSUE THREAD

**Issue by maurizio-lattuada** [linki/chaoskube#275]
In `README.md` is suggested to download Chaoskube's Helm chart from [Kubeapps](https://hub.kubeapps.com/charts/stable/chaoskube). Unfortunately, the chart there cannot be found: the HTTP GET https://hub.kubeapps.com/api/chartsvc/v1/charts/stable/chaoskube returns a 404 there.

Would it be possible for you to check whether the Helm chart is still published there? The old repository https://github.com/helm/charts/tree/master/stable/chaoskube is now in archived state since end November.

Thank you

**Comment by Perfect-Web** [comment]
same here. also any helm 3 version ?

**Comment by javajon** [comment]
This [Katacoda scenario that teaches chaoskube](https://katacoda.com/javajon/courses/kubernetes-chaos/chaoskube) relies on the Helm chart. The scenario will be updated/fixed once new chart location is made available.

**Comment by ghouscht** [comment]
Unfortunately I've the same issue here. Probably submitting the chart to the kuebapps Hub would be a good idea. I can help out with this if there is something missing which is "required" by the [best practices](https://hub.kubeapps.com/submit-repository) of kubeapps Hub.

**Comment by WyriHaximus** [comment]
The stable chart repository maintainer by Helm has been deprecated so that might be it. Is there an official Helm chart for Chaoskube?

**Comment by ghouscht** [comment]
I don't think so, no. But why not create one now? I can file a PR with the chart if @linki is ok with merging it.

**Comment by WyriHaximus** [comment]
That's the thing I was hinting at :+1: . Happy to help out

**Comment by ghouscht** [comment]
FYI: The PR was merged, you can now use the helm chart with:
```
helm repo add chaoskube https://linki.github.io/chaoskube
helm repo update
helm template chaoskube/chaoskube
```

**Comment by maurizio-lattuada** [comment]
Perfect, thank you all for this wonderful outcome :+1:

**Comment by javajon** [comment]
Tried this:

```bash
kubectl create namespace chaoskube

helm repo add chaoskube https://linki.github.io/chaoskube

helm repo update

helm install chaoskube chaoskube/chaoskube \
  --version 0.1.0 \
  --namespace chaoskube \
  --set 'tag=v0.21.0' \
  --set 'namespaces=!kube-system' \
  --set labels=app-purpose=chaos \
  --set interval=20s
```
Got this:

```bash
Events:
  Type     Reason     Age                    From               Message
  ----     ------     ----                   ----               -------
  Normal   Scheduled  4m28s                  default-scheduler  Successfully assigned chaoskube/chaoskube-87f854bc9-5dxsv to node01
  Normal   Pulling    2m54s (x4 over 4m26s)  kubelet, node01    Pulling image "quay.io/linki/chaoskube:0.21.0"
  Warning  Failed     2m52s (x4 over 4m25s)  kubelet, node01    Failed to pull image "quay.io/linki/chaoskube:0.21.0": rpc error: code = Unknown desc = Error response from daemon: manifest for quay.io/linki/chaoskube:0.21.0 not found: manifest unknown: manifest unknown
  Warning  Failed     2m52s (x4 over 4m25s)  kubelet, node01    Error: ErrImagePull
  Normal   BackOff    2m38s (x6 over 4m24s)  kubelet, node01    Back-off pulling image "quay.io/linki/chaoskube:0.21.0"
  Warning  Failed     2m27s (x7 over 4m24s)  kubelet, node01    Error: ImagePullBackOff
```

The Issue is image tag in repo is prefixed with "v" and this line strips the "v":

`image: "{{ .Values.image.repository }}:{{ .Values.image.tag | default .Chart.AppVersion }}"`

Also,

In value.yaml, the default for "tag" should be "latest", not "":

```bash
# Overrides the image tag whose default is the chart appVersion.
  tag: "latest"
```

Also, as part of the new chart for the merge, the install instructions in the README should have new Helm instructions.

**Comment by linki** [comment]
@javajon Thanks for trying and reporting this.

**Comment by linki** [comment]
@javajon I added a PR that fixed it for me: https://github.com/linki/chaoskube/pull/324

I don't think we should add `tag: "latest"` to the default values.yaml. This is only for when you want to override the default version defined in the Chart.yaml with something else. Let me know what you think about it.

**Comment by javajon** [comment]
It's best to peg it at a released version. The tag default is current blank
(""), which is a covert way of defining "latest". Suggested latest as
default to be self documenting. However, latest is an antipattern.

**Comment by linki** [comment]
@javajon I see, thanks for the explanation.

When I look at [the code](https://github.com/linki/chaoskube/blob/master/chart/chaoskube/templates/deployment.yaml#L28) it looks like given the `""` tag value it will use the chart's `appVersion` (the released version) rather than `latest`. Isn't that what we want by default?

**Comment by javajon** [comment]
With the default tag as "" then what version of Chaoskube will be used?
The missing value "" seems very indeterministic, not self-documenting, and
can easily break idempotency over time. Why not specify a precise recent
version of ChaosKube that the chart was tested with? If others want an
older or more recent version they can purposefully deviate from the
default. With a specific version, you are communicating that the chart has
been tested and verified with the specific version, such as  Chaoskube 0.20.0.
Good public charts tend to set the default tag to a definitive recent
version that the project sanctions as a working version with the chart. For
instance, the Bitnami group who support a RabbitMQ HA chart follow good
practice by specifying an exact tag for the recent RabbitMQ container tag:
3.8.14-debian-10-r0. See RabbitMQ default values for their chart.
<https://github.com/bitnami/charts/blob/master/bitnami/rabbitmq/values.yaml>

Also, IMHO the Chaoskube project public artifacts such as containers and
charts should not have a prefix of "v" as it confuses the issue and its
superfluous information in the tag. For instance, the redis containers
<https://hub.docker.com/_/redis?tab=tags&page=1&ordering=last_updated>
don't have the "v" prefix, like most other containers and charts.

   -

**Comment by javajon** [comment]
As part of this Helm chart installation update, it should be registered on the chaos list at [Artifacthub.io](https://artifacthub.io/packages/search?ts_query_web=chaos).
